File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

    iex> RnaTranscription.to_rna(~c"ACTG")
    ~c"UGAC"
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    Enum.map(dna, &transcribe/1)
  end

  defp transcribe(71), do: 67
  defp transcribe(67), do: 71
  defp transcribe(84), do: 65
  defp transcribe(65), do: 85
end

File: dart/rna-transcription/lib/rna_transcription.dart
class RnaTranscription {
  String toRna(String dna) {
    List<String> dnaParts = dna.split('');
    List<String> rnaParts = dnaParts.map((d) => transcribe(d)).toList();
    return rnaParts.join('');
  }

  String transcribe(String d) {
    String res = '';
    switch (d) {
      case 'G':
        res = 'C';
        break;
      case 'C': 
        res = 'G';
        break;
      case 'T': 
        res = 'A';
        break;
      case 'A': 
        res = 'U';
        break;
    }
    return res;
  }
}
